src/xunit.performance.api/Table/CSVFile.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+namespace Microsoft.Xunit.Performance.Api.Table
+{
+    internal sealed class CSVReader : IDisposable
+    {
+        private StreamReader _Reader;
+        private Dictionary<string, int> titlePositions = new Dictionary<string, int>(StringComparer.OrdinalIgnoreCase);
+        private List<string[]> lines = new List<string[]>();
+
+        public CSVReader(string filePath)
+        {
+            var fs = new FileStream(filePath, FileMode.Open);
+            this._Reader = new StreamReader(fs);
+
+            // Get the title line
+            string line = this._Reader.ReadLine();
+            if (line == null)
+            {
+                throw new InvalidDataException("No lines in CSV file.");
```

Comment:
Does `Dispose()` get automatically called here? If not, this throw is leaking `_Reader`, which was opened above but doesn't get closed.